### PR TITLE
Add NATS credentials file authentication

### PR DESCRIPTION
**Pull request by ddecaro11**
# Description

Allow NATS JS streaming source to authenticate through a credentials file

# How Has This Been Tested?

Performed integration tests on a jwt enabled NATS server

# Checklist
- [ X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [ X] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works ###not unit testable, only integration test possible
- [ X] I have commented my code, particularly in hard-to-understand areas
- [ X] I have made corresponding changes to the documentation

cc:

@wangxiaoyou1993

### CODE REVIEW

**Review by wangxiaoyou1993 — APPROVED** [review]

File: mage_ai/data_preparation/templates/data_loaders/streaming/nats.yaml [hunk 1]
```diff
@@ -24,5 +24,8 @@ use_tls: false
 # Optional: NKEY seed string
 # nkeys_seed_str: "SUAPAEYJQOQHJ4"
 
+# Optional: user credentials
+# user_credentials: "path/to/credentials.creds"
```
Comment (trover97) [inline mage_ai/data_preparation/templates/data_loaders/streaming/nats.yaml:28]: Как выглядит это файл?

Comment (ddecaro11) [reply mage_ai/data_preparation/templates/data_loaders/streaming/nats.yaml:28]: It's a classic NATS credential format, an example below:

-----BEGIN NATS USER JWT-----
eyJ0eXAiOiJKV1QiLCJhbGciOiJlZDI1NTE5LW5rZXkifQ.eyJqdGkiOiJURjRLNk41NFQzRTRNT1kyNkRDSVVQRjNUSFJUQ05KRENENU1OWk82WERHUDJFSUFUVjVBIiwiaWF0IjoxNzIwMTg3OTUwLCJpc3MiOiJBQ0FXSlVPNk5aVUpZVU9JTVVGUUJZUVpXWE5UUERMM0NDNFBaWE1SS083T1dPTUs2WU0ySFY0QSIsInN1YiI6IlVETkRNWjRGRUdEWDRIUkNNWk82VVlCSk1IUFI0UDM1NDMzTVcyV1U3WDQ3NUlCTDNGS0dNMlZNIiwibmF0cyI6eyJwdWIiOnsiYWxsb3ciOlsiZGV2aWNlcy45MDUzYzk4Yy01NDNkLTQzMmQtODNjOS1lMjM0OTJiMDJkNGMuXHUwMDNlIiwiJEtWLmRldmljZV9jb25maWcuZGV2aWNlcy45MDUzYzk4Yy01NDNkLTQzMmQtODNjOS1lMjM0OTJiMDJkNGMuY29uZmlnLmFwcGxpZWQiLCIkSlMuQVBJLlNUUkVBTS5JTkZPLktWX2RldmljZV9jb25maWciLCIkSlMuQVBJLkNPTlNVTUVSLkNSRUFURS5LVl9kZXZpY2VfY29uZmlnLiouJEtWLmRldmljZV9jb25maWcuZGV2aWNlcy45MDUzYzk4Yy01NDNkLTQzMmQtODNjOS1lMjM0OTJiMDJkNGMuXHUwMDNlIiwiJEpTLkFQSS5DT05TVU1FUi5ERUxFVEUuS1ZfZGV2aWNlX2NvbmZpZy4qIiwiJEpTLkFQSS5ESVJFQ1QuR0VULktWX2RldmljZV9jb25maWcuJEtWLmRldmljZV9jb25maWcuZGV2aWNlcy45MDUzYzk4Yy01NDNkLTQzMmQtODNjOS1lMjM0OTJiMDJkNGMuXHUwMDNlIl19LCJzdWIiOnsiYWxsb3ciOlsiZGV2aWNlcy45MDUzYzk4Yy01NDNkLTQzMmQtODNjOS1lMjM0OTJiMDJkNGMuXHUwMDNlIiwiX0lOQk9YLlx1MDAzZSJdfSwic3VicyI6MTAsImRhdGEiOi0xLCJwYXlsb2FkIjoxMDI0MDAsImlzc3Vlcl9hY2NvdW50IjoiQURaNFRWVzJGR0pXSFRTUVVZU1FVTzI0QlBTUjY2SlpGSVlKSVhHVDdLTzJRRVNFTVhEVUNJSVAiLCJ0eXBlIjoidXNlciIsInZlcnNpb24iOjJ9fQ.f0tqVJVltjHJZSxBKZ4_PWHw8fGtcAIm3SXNUuXg1epw3rj_yQwIv-Dd0kOBIrxlmHVtrJ5DjuOvYMwG_Nc0BA
------END NATS USER JWT------

************************* IMPORTANT *************************
NKEY Seed printed below can be used to sign and prove identity.
NKEYs are sensitive and should be treated as secrets.

-----BEGIN USER NKEY SEED-----
SUAD5MKX35HKDNXDXLJHQZA6XD7QGHH54SXERM3FPDRVJSXK43RNSJZX7Q
------END USER NKEY SEED------

*************************************************************

Comment (trover97) [reply mage_ai/data_preparation/templates/data_loaders/streaming/nats.yaml:28]: Is it only jwt authorization? How can I use the user/pass format?

Comment (ddecaro11) [reply mage_ai/data_preparation/templates/data_loaders/streaming/nats.yaml:28]: This PR adresses only JWT auth. For user/password other properties should be forwarded to nats client from yaml configuration, I don't think it has been implemented yet

Comment (trover97) [reply mage_ai/data_preparation/templates/data_loaders/streaming/nats.yaml:28]: Ok, thanks)

**Pull request merged**